presentation/common/src/presentation-common/content/PropertyValueFormatter.ts

```diff
@@ -120,76 +120,89 @@ export class ContentPropertyValueFormatter {
-    switch (type.valueFormat) {
+  private async formatValue(field: Field, value: Value, ctx?: { doubleFormatter: (raw: number) => Promise<string> }): Promise<DisplayValue> {
+    switch (field.type.valueFormat) {
       case PropertyValueFormat.Primitive:
-        return this.formatPrimitiveValue(type, value, ctx);
+        return this.formatPrimitiveValue(field, value, ctx);
       case PropertyValueFormat.Array:
-        return this.formatArrayValue(type, value);
+        return this.formatArrayValue(field.type, value, field);
       case PropertyValueFormat.Struct:
-        return this.formatStructValue(type, value);
+        return this.formatStructValue(field.type, value, field);
     }
   }
 
-  private async formatPrimitiveValue(type: PrimitiveTypeDescription, value: Value, ctx?: { doubleFormatter: (raw: number) => Promise<string> }) {
+  private async formatPrimitiveValue(field: Field, value: Value, ctx?: { doubleFormatter: (raw: number) => Promise<string> }) {
     if (value === undefined) {
       return "";
     }
 
     const formatDoubleValue = async (raw: number) => (ctx ? ctx.doubleFormatter(raw) : formatDouble(raw));
 
-    if (type.typeName === "point2d" && isPoint2d(value)) {
+    if (field.type.typeName === "point2d" && isPoint2d(value)) {
       return `X: ${await formatDoubleValue(value.x)}; Y: ${await formatDoubleValue(value.y)}`;
     }
-    if (type.typeName === "point3d" && isPoint3d(value)) {
+    if (field.type.typeName === "point3d" && isPoint3d(value)) {
       return `X: ${await formatDoubleValue(value.x)}; Y: ${await formatDoubleValue(value.y)}; Z: ${await formatDoubleValue(value.z)}`;
     }
-    if (type.typeName === "dateTime") {
+    if (field.type.typeName === "dateTime") {
       assert(typeof value === "string");
       return value;
     }
-    if (type.typeName === "bool" || type.typeName === "boolean") {
+    if (field.type.typeName === "bool" || field.type.typeName === "boolean") {
       assert(typeof value === "boolean");
       return value ? "@Presentation:value.true@" : "@Presentation:value.false@";
     }
-    if (type.typeName === "int" || type.typeName === "long") {
+    if (field.type.typeName === "int" || field.type.typeName === "long") {
       assert(isNumber(value));
       return value.toFixed(0);
     }
-    if (type.typeName === "double") {
+    if (field.type.typeName === "double") {
       assert(isNumber(value));
       return formatDoubleValue(value);
     }
-    if (type.typeName === "navigation") {
+    if (field.type.typeName === "navigation") {
       assert(Value.isNavigationValue(value));
       return value.label.displayValue;
     }
 
+    if (field.type.typeName === "enum" && field.isPropertiesField()) {
+      return field.properties[0].property.enumerationInfo?.choices.find(({ value: enumValue }) => enumValue === value)?.label;
```

Comment:
Enumerations may have `isStrict = false`, which should allow them to receive values other than the ones in `choices` list. So if the enum is not strict and we don't find a choice, we should return raw value.

Reply:
Changed